src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -4431,7 +4438,8 @@ protected Object singleByteOptimizable(Rope stringRope, Rope patternRope, int of
 
         @TruffleBoundary
         @Specialization(
-                guards = "!singleByteOptimizableNode.execute(stringRope)")
+                guards = "!singleByteOptimizableNode.execute(stringRope, " +
+                        "getContext().getEncodingManager().getRubyEncoding(stringRope.encoding.getIndex()))")
```

Comment:
This should be replaced by passing the RubyEncoding explicitly if possible (for efficiency & propagating the correct encoding).